numpy/core/src/multiarray/conversion_utils.c

```diff
@@ -553,6 +557,13 @@ PyArray_OrderConverter(PyObject *object, NPY_ORDER *val)
     }
     else {
         str = PyBytes_AS_STRING(object);
+        if (strlen(str) != 1) {
+            DEPRECATE("Non length-one string passed in " \
```

Comment:
I'm sorry: could you clarify what you mean by this?  I tried reading the documentation on <a href="https://docs.python.org/">Python Docs</a>, but I'm not sure I quite follow.